src/connectionController.ts

```diff
@@ -240,140 +297,92 @@ export default class ConnectionController {
     }
   }
 
-  public sendTelemetry(
-    newDataService: DataServiceType,
-    connectionModel: ConnectionModel,
-    connectionType: ConnectionTypes
-  ): void {
-    void this._telemetryService.trackNewConnection(
-      newDataService.client.client,
-      connectionModel,
-      connectionType
-    );
+  public sendTelemetry(newDataService: DataService, connectionType: ConnectionTypes): void {
+    void this._telemetryService.trackNewConnection(newDataService, connectionType);
   }
 
-  parseNewConnection(newConnectionModel: ConnectionModel): ConnectionModel {
-    // Here we re-parse the connection, as it can be loaded from storage or
-    // passed by the connection model without the class methods.
-    const connectionModel: ConnectionModel = new Connection(
-      newConnectionModel
-    );
+  parseNewConnection(rawConnectionModel: RawConnectionModel): ConnectionInfo {
+    const connectionModel = new ConnectionModel(rawConnectionModel);
```

Comment:
I guess this is the part where we’re not sure about whether it’s alright to keep going with `ConnectionModel`? I would think it’s okay, yeah. We should probably just leave a ticket about _eventually_ removing it. (But it might be nice if @Anemy could confirm this :slightly_smiling_face:)

Reply:
I think it's fine yup. It's probably not worth adding all the logic to build the connectionInfo from the connection configuration object the form in the webview creates. Would be nice to remove this usage of connection-model eventually, and this would definitely be something that sharing the connect form would help make go away as well.